client/mussel/v12.03.d/instance.sh

```diff
@@ -132,6 +132,37 @@ task_show_password() {
   call_api -X GET $(base_uri)/${namespace}s/${uuid}/password.$(suffix)
 }
 
+task_decrypt_password() {
+  local namespace=$1 cmd=$2 uuid=$3 ssh_key_pair_path=$4
+  [[ -n "${namespace}" ]] || { echo "[ERROR] 'namespace' is empty (${BASH_SOURCE[0]##*/}:${LINENO})" >&2; return 1; }
+  [[ -n "${cmd}"       ]] || { echo "[ERROR] 'cmd' is empty (${BASH_SOURCE[0]##*/}:${LINENO})" >&2; return 1; }
+  [[ -n "${uuid}"      ]] || { echo "[ERROR] 'uuid' is empty (${BASH_SOURCE[0]##*/}:${LINENO})" >&2; return 1; }
+  [[ -f "${ssh_key_pair_path}" ]] || { echo "[ERROR] 'ssh_key_file' is incorrect (${BASH_SOURCE[0]##*/}:${LINENO})" >&2; return 1; }
+
+  instance_password="$(task_show_password "$namespace" "$cmd" $uuid)" \
+      || { echo "[ERROR] encrypted password not available (${BASH_SOURCE[0]##*/}:${LINENO})" >&2; return 1; }
+  # ---
+  # :id: i-czulsyoj
+  # :encrypted_password: |
+  #  IAAMQn+4axLf/EO+dzQWhzDOBY/dkOCvDegiL8eS1VTs9MrwgA05vnCQ//EO
+  #  RaZPfWQLV4qhIjr1h4RUNQZ41Hs22lOztb2qpACuQzlfXpTKDp3YhZdOw/V8
+  #  hencQ02g8uziV5+Wpy1DLikkATMrRLurp2zsg7XQp/0mO9e16YzkzO9Xor3R
+  #  C+daGS/YeB8BZqQbwZgzWLnrgRm7q6zzKJoGxl67+RYHki/19n1gFQXNVyOi
+  #  /gZo1sdAPF/N2a18naOM4lpbjRr4R2u82mkgCMFYzRJew9yvOTGkFhwZamkX
+  #  9j+wU5Cvo0fDDpS4pXlRTvbmu9ESTmg0tSfFyuCgkA==
+
+  # assume :encrypted_password: is the last field and
```

Comment:
Yaml/json parsing is a weak spot for bash. I like the idea of getting some dedicated code for it somewhere in mussel but until then I think (1) is the way to go. Having a single awk solution that's different from everywhere else in the good doesn't sound like a good idea to me.